### PR TITLE
Add rule for checking selinux is not disabled in coreos

### PR DESCRIPTION
**Pull request by JAORMX**
This also enables the `oval_argument_value_in_line_criterion` template
to be negated and it adds to the `coreos_kernel_option` OVAL template as well.

### CODE REVIEW

**Comment by yuumasato** [comment]
@openscap-ci test this please

**Comment by evgenyz** [comment]
Looks good. @yuumasato what do you think about the template argument name `negate`?

**Review by evgenyz — APPROVED** [review]

File: linux_os/guide/system/selinux/coreos_enable_selinux_kernel_argument/rule.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <tt>MachineConfig</tt> object.
+
+template:
+    name: coreos_kernel_option
+    vars:
+        arg_name: selinux
+        arg_value: '0'
+        arg_negate: 'true'
+    backends:
+        kubernetes: "off"
```
Comment (ggbecker) [inline linux_os/guide/system/selinux/coreos_enable_selinux_kernel_argument/rule.yml:52]: please add a new line here

File: shared/templates/coreos_kernel_option/kubernetes.template [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{{%- if not ARG_NEGATE %}}
```
Comment (ggbecker) [inline shared/templates/coreos_kernel_option/kubernetes.template:1]: I don't think we need this macro, the `backends` should do trick. Or do you still want to have this?

File: linux_os/guide/system/selinux/grub2_enable_selinux/rule.yml [hunk 3]
```diff
@@ -20,7 +20,6 @@ severity: medium
 identifiers:
     cce@rhel7: CCE-26961-3
     cce@rhel8: CCE-80827-9
-    cce@rhcos4: CCE-82666-9
```
Comment (ggbecker) [inline linux_os/guide/system/selinux/grub2_enable_selinux/rule.yml:23]: Was this rule ever shipped in a release? If yes, I don't think it makes sense to remove the CCE even if it's not part of a profile anymore.

**Review by ggbecker — CHANGES_REQUESTED** [review]
For the new template argument `arg_negate`, we have to include a pre-processing exactly like this: https://github.com/ComplianceAsCode/content/blob/master/shared/templates/sshd_lineinfile/template.py otherwise it would fail to detect `false` parameter.

I also notice that the template `coreos_kernel_option` misses documentation in the official devel doc page. It would probably go here: https://github.com/ComplianceAsCode/content/blame/master/docs/manual/developer/06_contributing_with_content.md#L1270

File: shared/templates/coreos_kernel_option/kubernetes.template (hunk 2 above)
Comment (JAORMX) [reply shared/templates/coreos_kernel_option/kubernetes.template:1]: nope, let me remove it.

File: linux_os/guide/system/selinux/grub2_enable_selinux/rule.yml (hunk 3 above)
Comment (JAORMX) [reply linux_os/guide/system/selinux/grub2_enable_selinux/rule.yml:23]: OK, I'll change it.

File: linux_os/guide/system/selinux/coreos_enable_selinux_kernel_argument/rule.yml (hunk 1 above)
Comment (JAORMX) [reply linux_os/guide/system/selinux/coreos_enable_selinux_kernel_argument/rule.yml]: Done

**Comment by JAORMX** [comment]
> For the new template argument `arg_negate`, we have to include a pre-processing exactly like this: https://github.com/ComplianceAsCode/content/blob/master/shared/templates/sshd_lineinfile/template.py otherwise it would fail to detect `false` parameter.
> 
> I also notice that the template `coreos_kernel_option` misses documentation in the official devel doc page. It would probably go here: https://github.com/ComplianceAsCode/content/blame/master/docs/manual/developer/06_contributing_with_content.md#L1270

Done

**Comment by JAORMX** [comment]
/test e2e-aws-ocp4-cis

remediation timed out. Unrelated to this.

**Comment by JAORMX** [comment]
this passed the e2e test:
```
    helpers.go:676: Result - Name: e2e-moderate-master-coreos-enable-selinux-kernel-argument - Status: PASS - Severity: medium
    helpers.go:747: Rule coreos_enable_selinux_kernel_argument matched expected result
```

File: docs/manual/developer/06_contributing_with_content.md [hunk 4]
```diff
@@ -1268,6 +1268,21 @@ the following to `rule.yml`:
...
+-   Checks that `argument=value` pair is present in the kernel arguments.
+    Note that this applies to Red Hat CoreOS.
+
+-   Parameters:
+
+    -   **arg_name** - Argument name, eg. `audit`
+
+    -   **arg_value** - Argument value, eg. `'1'`
+
+    -   **negate** - negates the check, which then ensures that
```
Comment (ggbecker) [inline docs/manual/developer/06_contributing_with_content.md:1281]:
```suggestion
    -   **arg_negate** - negates the check, which then ensures that
```

The actual parameter name is `arg_negate` although I don't know which of them makes more sense, probably something more verbose like `negate_check` or something

Comment (ggbecker) [reply docs/manual/developer/06_contributing_with_content.md:1281]: But I'm fine with `arg_negate` as well

**Review by ggbecker — CHANGES_REQUESTED** [review]
A minor issue in the documentation. Other than that, it looks good.

**Review by ggbecker — APPROVED** [review]
LGTM. There are some tests still running but they were verified to be working before the last commit, no need to wait since the last commit only addresses one minor issue in the text docs.

**Pull request merged**